tools/monorepo-utils/src/code-freeze/commands/changelog/lib/index.ts

```diff
@@ -0,0 +1,187 @@
+/**
+ * External dependencies
+ */
+import simpleGit from 'simple-git';
+import { execSync } from 'child_process';
+
+/**
+ * Internal dependencies
+ */
+import { Logger } from '../../../../core/logger';
+import { checkoutRemoteBranch } from '../../../../core/git';
+import { createPullRequest } from '../../../../core/github/repo';
+import { Options } from '../types';
+
+/**
+ * Perform changelog operations on release branch by submitting a pull request. The release branch is a remote branch.
+ *
+ * @param {Object} options       CLI options
+ * @param {string} tmpRepoPath   temp repo path
+ * @param {string} releaseBranch release branch name. The release branch is a remote branch on Github.
+ * @return {Object} update data
+ */
+export const updateReleaseBranchChangelogs = async (
+	options: Options,
+	tmpRepoPath: string,
+	releaseBranch: string
+): Promise< { deletionCommitHash: string; prNumber: number } > => {
+	const { owner, name, version } = options;
+	try {
+		await checkoutRemoteBranch( tmpRepoPath, releaseBranch );
+	} catch ( e ) {
+		if ( e.message.includes( "couldn't find remote ref" ) ) {
+			Logger.warn(
+				`${ releaseBranch } does not exist on ${ owner }/${ name }.`
+			);
+		}
+		Logger.error( e );
+	}
+
+	const git = simpleGit( {
+		baseDir: tmpRepoPath,
+		config: [ 'core.hooksPath=/dev/null' ],
+	} );
+
+	const branch = `update/${ version }-changelog`;
+
+	await git
+		.checkout( {
+			'-b': null,
+			[ branch ]: null,
+		} )
+		.catch( ( e ) => {
+			Logger.error( e );
+		} );
+
+	Logger.notice( `Running the changelog script in ${ tmpRepoPath }` );
+	execSync(
+		`pnpm --filter=woocommerce run changelog write --add-pr-num -n -vvv --use-version ${ version }`,
+		{
+			cwd: tmpRepoPath,
+			stdio: 'inherit',
+		}
+	);
+
+	//Checkout pnpm-lock.yaml to prevent issues in case of an out of date lockfile.
+	await git.checkout( 'pnpm-lock.yaml' ).catch( ( e ) => {
+		Logger.error( e );
+	} );
+
+	Logger.notice( `Committing deleted files in ${ tmpRepoPath }` );
+
+	await git.add( 'plugins/woocommerce/changelog/' ).catch( ( e ) => {
+		Logger.error( e );
+	} );
+	await git
+		.commit( `Delete changelog files from ${ version } release` )
+		.catch( ( e ) => {
+			Logger.error( e );
+		} );
+
+	// Can use git.revparse() here I think.
+	const deletionCommitHash = await git.raw( [ 'rev-parse', 'HEAD' ] );
+	Logger.notice( `git deletion hash: ${ deletionCommitHash }` );
+
+	// Do readme.txt changes
+	Logger.notice( `Updating readme.txt in ${ tmpRepoPath }` );
+	execSync( 'php .github/workflows/scripts/release-changelog.php', {
+		cwd: tmpRepoPath,
+		stdio: 'inherit',
+	} );
```

Comment:
This is calling a PHP script. I'm leaving it here for now, but will tackle converting to TS in a follow up.